### PR TITLE
fix android crash while use vulkan.  game logic code may add lable while enter background/foreground

### PR DESCRIPTION
**Pull request by bofeng-song**
…e enter background/foreground

(cherry picked from commit 3481b5a5e2db4618ea8f42ba807e876ddaa8fbe8)

Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp [hunk 1]
```diff
@@ -309,8 +309,6 @@ bool CCVKSwapchain::checkSwapchainStatus(uint32_t width, uint32_t height) {
 
     destroySwapchain(gpuDevice);
 
-    gpuDevice->curBackBufferIndex = 0;
```
Comment (bofeng-song) [inline native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp:312]: While enter background, and then back to foreground. Some cases may add label node, it`ll trigger create texture buffers, and texture buffers bind to the index. After call doCreateSurface, texture buffers`s index maybe bind to the wrong index.

**Comment by pandamicro** [comment]
@star-e please evaluate whether it's possible and safe to rebase to v3.6

**Review by star-e — APPROVED** [review]

**Comment by star-e** [comment]
> @star-e please evaluate whether it's possible and safe to rebase to v3.6

I think this pr should be ported to v3.6 and has to be tested before merging.

File: native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp (hunk 1 above)
Comment (bofeng-song) [reply native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp:312]: The real cause of the crash is that the _buffersToBeUpdated cache instruction is used in the code, and the stored and used curBackBufferIndex may be modified

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
It is merged in v3.6.0.

**Pull request closed without merging**